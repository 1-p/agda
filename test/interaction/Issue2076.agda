-- Andreas, 2016-07-06, issue #2076 reported by Andrea
-- This is to test printing of extended lambdas

-- {-# OPTIONS -v tc.cc:60 -v tc.term.exlam:100 -v reify.clause:60 #-}

postulate
  A : Set
  _,_ : A → A → A

data Unit : Set where
  true : Unit

bar : (p : A) (q : A) → Unit → A
bar p q = aux
  where
  aux : Unit → A
  aux true = p , q

foo : (p : A) (q : A) → Unit → A
foo p q = \ { true → p , q}

test : (a : A) → Unit → A
test a = foo a a
-- Normalize me! Expected:
-- λ a → λ { true → a , a }

-- From test/interaction/ExtendedLambdaCase.agda

data Bool : Set where
  true false : Bool

data Void : Set where

module M {A : Set}(B : A → Set) where
  postulate
    Bar : (Bool → Bool) → Set

Test : Set
Test = (y : Bool) → M.Bar {Bool} (λ _ → Void) (λ { true → true ; false → false })
-- Normalize me! Expected:
-- (y : Bool) → M.Bar (λ _ → Void) (λ { true → true ; false → false })

-- .extendedlambda1 y true  = true
-- .extendedlambda1 y false = false
-- dropped args:  []
-- hidden  args:  []
-- visible args:  [y]
