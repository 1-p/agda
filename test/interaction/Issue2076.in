top_command (cmd_load currentFile [])
top_command (cmd_compute_toplevel False "test")
top_command (cmd_compute_toplevel False "Test")
