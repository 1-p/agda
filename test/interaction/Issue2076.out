(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue2076 (Issue2076.agda). " t)
(agda2-info-action "*Type-checking*" "Finished Issue2076. " t)
(agda2-status-action "Checked")
(agda2-info-action "*All Goals*" "" nil)
((last . 1) . (agda2-goals-action '()))
(agda2-status-action "Checked")
(agda2-info-action "*Normal Form*" "λ a → λ { true → a , a }" nil)
((last . 1) . (agda2-goals-action '()))
(agda2-status-action "Checked")
(agda2-info-action "*Normal Form*" "(y : Bool) → M.Bar (λ _ → Void) (λ { true → true ; false → false })" nil)
((last . 1) . (agda2-goals-action '()))
